flink-connectors/flink-connector-gcp-pubsub/src/test/java/org/apache/flink/streaming/connectors/gcp/pubsub/PubSubSourceTest.java

```diff
@@ -66,13 +65,17 @@ public class PubSubSourceTest {
 
     private PubSubSource<String> pubSubSource;
 
-    @Before
-    public void setup() throws Exception {
-        when(pubSubSubscriberFactory.getSubscriber(eq(credentials))).thenReturn(pubsubSubscriber);
-        when(streamingRuntimeContext.isCheckpointingEnabled()).thenReturn(true);
-        when(streamingRuntimeContext.getMetricGroup()).thenReturn(metricGroup);
-        when(metricGroup.addGroup(any(String.class))).thenReturn(metricGroup);
-        when(acknowledgeOnCheckpointFactory.create(any())).thenReturn(acknowledgeOnCheckpoint);
+    @BeforeEach
+    void setup() throws Exception {
+        lenient()
+                .when(pubSubSubscriberFactory.getSubscriber(eq(credentials)))
+                .thenReturn(pubsubSubscriber);
+        lenient().when(streamingRuntimeContext.isCheckpointingEnabled()).thenReturn(true);
+        lenient().when(streamingRuntimeContext.getMetricGroup()).thenReturn(metricGroup);
+        lenient().when(metricGroup.addGroup(any(String.class))).thenReturn(metricGroup);
+        lenient()
+                .when(acknowledgeOnCheckpointFactory.create(any()))
+                .thenReturn(acknowledgeOnCheckpoint);
```

Comment:
I know that there's a bit of time passed since you touched this PR. But can you elaborate a bit on why we're switching to `lenient` here rather than sticking to the old approach. My understanding is that `lenient()` relaxes the test condition a bit?

Reply:
It looks like these stubbing methods are not always required for all the tests
e.g. `when(pubSubSubscriberFactory.getSubscriber(eq(credentials)))
                .thenReturn(pubsubSubscriber)` is required only for `testOpenWithCheckpointing`
for others it fails with `Unnecessary stubbings detected.`
Similar for others. It is strange to me why it didn't fail like that with JUnit4

Reply:
After some research it seems that in case of JUnit5 Mockito works a bit different in compare how it works for JUnit4.
The reason is that for JUnit5 there `org.mockito.junit.jupiter.MockitoExtension` [1] is used
and for JUnit4 `org.mockito.junit.MockitoJUnitRunner` [2].

The way how validation of used stubs is working is also different.
E.g. in case of JUnit5 it is handled at `org.mockito.MockitoSession#finishMocking(java.lang.Throwable)` [3] which is invoked after each test.
At the same time in JUnit4 it is processed inside `org.mockito.internal.runners.StrictRunner#run` [4] and logic there is the next one: 
first it works only for cases when all tests from class/testsuite are running
second it does validation of used/not used stubs after all tests finished.


Seems it is the reason why after movement to JUnit5 it starts failing
[1] https://github.com/mockito/mockito/blob/v3.4.6/subprojects/junit-jupiter/src/main/java/org/mockito/junit/jupiter/MockitoExtension.java
[2] https://github.com/mockito/mockito/blob/v3.4.6/src/main/java/org/mockito/junit/MockitoJUnitRunner.java
[3] https://github.com/mockito/mockito/blob/v3.4.6/subprojects/junit-jupiter/src/main/java/org/mockito/junit/jupiter/MockitoExtension.java#L184-L186
[4] https://github.com/mockito/mockito/blob/v3.4.6/src/main/java/org/mockito/internal/runners/StrictRunner.java#L45-L53

Reply:
Thanks for clarification, @snuyanzin